src/zap/zapinfo.cpp

```diff
@@ -3681,7 +3699,28 @@ unsigned ZapInfo::getMethodHash(CORINFO_METHOD_HANDLE ftn)
 
 DWORD ZapInfo::getMethodAttribs(CORINFO_METHOD_HANDLE ftn)
 {
-    return m_pEEJitInfo->getMethodAttribs(ftn);
+    DWORD result = m_pEEJitInfo->getMethodAttribs(ftn);
+
+#if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
+    if (result & CORINFO_FLG_JIT_INTRINSIC)
+    {
+        // Do not report the get_IsSupported method as an intrinsic. This will turn the call into a regular
+        // call. We also make sure none of the hardware intrinsic method bodies get pregenerated in crossgen
+        // (see ZapInfo::CompileMethod) but get JITted instead. The JITted method will have the correct
+        // answer for the CPU the code is running on.
+
+        const char* namespaceName;
+        const char* methodName = getMethodNameFromMetadata(m_currentMethodHandle, nullptr, &namespaceName, nullptr);
+
+        if (strcmp(methodName, "get_IsSupported") == 0
```

Comment:
Technically, we could allow `IsSupported` checks for SSE and SSE2 to be intrinsic, but I didn't want to complicate this more than necessary. I can add that.